pkg/ipcache/ipcache.go

```diff
@@ -633,10 +644,14 @@ func (ipc *IPCache) deleteLocked(ip string, source source.Source) (namedPortsCha
 		namedPortsChanged = ipc.updateNamedPorts()
 	}
 
+	if newHostIP != nil {
+		nodeID = ipc.AllocateNodeID(newHostIP)
```

Comment:
@pchaigno why does `deleteLocked()` allocate a new node ID rather than releasing the node ID?

Reply:
Is it possible to leak or exhaust the node ID space? In some environments, users may churn nodes on a regular basis.